react.md

```diff
@@ -1,90 +1,134 @@
+
+ This design pattern is good because it provides separation of concerns. Designers can work on the UI Dumb Component without the need to know all the logic. It also makes the component more reusable, as you can pass different logic to the same component or pass the same logic to different components.
+
+
+
+# 3. When we use 'yarn add ...' in the terminal - what is yarn doing? And what file will always be automatically updated after we add a package with yarn?
+
+ // Your Answer
+
+ Yarn is a program/application installation manager, so 'yarn add' begins this process. So far we've used it for creating React Apps.
+
+ // Googled Answer
+
+ 'Yarn Install' is used to install all dependencies for a project. This is most commonly used when you have just checked out code for a project. This command has been replaced by 'yarn add'.
+
+ This will also update your package.json and your yarn.lock so that other developers working on the project will get the same dependencies as you when they run yarn or yarn install.
+
+
+
+# 4. ???
+
+
+
+# 5. There are three mistakes in this code that would cause it to break our application. Find the mistakes and fix them:
+
+    import React, { Component } from 'react'
+
+    class Recipes extends Component {
+      constructor(props) {
         super(props)
         this.state = {
-          recipes: 
+          recipes:
             {name: 'Meatballs'},
             {name: 'Mac & Cheese'}
-      
         }
       }
 
       render() {
-    
+
+        let recipes = this.state.recipes.map(function(recipe) {
+          return (
+            <li key = {recipe.name} > {recipe.name} </li>
+        })
+
         return (
-    
-          let recipes = this.state.recipes.map(function(recipe){
-            return(
-              <li key={recipe.name}>{recipe.name}</li>
-            )
-          })
-    
+
           <ul>
             {recipes}
           </ul>
-        );
+
+        )
       }
     }
 
-    export default Recipes;
-
-#### 6. Name three html input types. (NOTE: text is the default type - so it doesn't count in this case)
- 
- //Your Answer
- 
- 
- //Googled Answer
- 
- 
- #### 7. How would you explain state to a friend who doesn't know code?
- 
- //Your Answer
- 
- 
- //Googled Answer
- 
- 
- #### 8. What is the difference between component state and props? Your answer should include a short explanation of both.
- 
- 
- //Your Answer
- 
- 
- //Googled Answer
- 
-   
-#### 9. Write a paragraph or so about your experience with building tic-tac-toe. Some topics to start with might be: things you learned about yourself, concepts from React that stood out to you, something about pair programming (if you paired), or the experience of building something in code from scratch.
+    export default Recipes
+
+
+
+# 6. Name three html input types. (NOTE: text is the default type - so it doesn't count in this case)
```

Comment:
You found some good examples in your search. There are quite a few others. Go ahead and update your response with some others.